main.js

```diff
@@ -0,0 +1,142 @@
+const BoardState = function () {
+  let board = [];
+
+  let discs = 3;
+
+  const initializeGame = function (number) {
+    if (number) {
+      discs = number;
+    } 
+
+    let newBoard = new Array(discs);
+
+    let num = discs;
+
+    // update board array
+    newBoard = newBoard.fill(0).map((element) => {
+      let value = element + num;
+      num--;
+      return value;
+    });
+
+    board = [newBoard, [], []];
+
+    // display board
+    console.log(
+      `Objective: Move a stack of disks from the top peg to the bottom peg, following the rules that only one disk can be moved at a time and no disk can be placed on top of a smaller disk.\nTo move a disk, type "moveDisc(fromPeg, toPeg)" and replace "fromPeg" with a number between 1 (top peg) and 3 (bottom peg) and "toPeg" with a number between 1 and 3.\nTo increase the difficulty, use the "setDiscs(number)" function and replace "number" with the desired number of discs (at least 3, at most 12). Note: Although you are allowed to add 12 discs, it would take you 2^12 - 1 (4,095) steps to complete. If you want to watch the computer solve the puzzle for you, type "runAlgorithm()". \n\n` + displayBoard()
+    );
+
+  };
+
+  const update = function (fromPeg, toPeg) {
+    board[toPeg].push(board[fromPeg].pop());
+  };
+
+  const displayBoard = function () {
+    const returnBoard = board.map((element) => {
+      return `---  ${element} \n`;
+    });
+
+    return returnBoard.toString()
+    .replaceAll("\n,", "\n")
+    .replaceAll(",", " ");
+  };
+
+  const logMove = function (fromPeg, toPeg) {
+    // Check if move is legal
+    const prevBoard = displayBoard();
+
+    if (fromPeg < 0 || fromPeg > 2 || toPeg < 0 || toPeg > 2) {
```

Comment:
would be more DRY (don't repeat yourself principal) to create a string and then do a bunch of if/else if to figure which string to log